python/pyspark/sql/functions.py

```diff
@@ -9929,6 +10078,26 @@ def map_zip_with(
     return _invoke_higher_order_function("MapZipWith", [col1, col2], [f])
 
 
+def str_to_map(
+    col1: "ColumnOrName",
+    col2: Union["ColumnOrName", str],
+    col3: Union["ColumnOrName", str],
+) -> Column:
```

Comment:
Because similar problems also exist here, @zhengruifeng 
😄

Reply:
thank you so much! @panbingkun 

since `ColumnOrName = Union[Column, str]`  https://github.com/apache/spark/blob/32ab341071aa69917f820baf5f61668c2455f1db/python/pyspark/sql/_typing.pyi#L37

we will always treat a str var as a column name in this case, if user want to provide a literal string, he will have to use `lit(some_string)`

Reply:
so `col2` and `col3` should be "ColumnOrName", and if the input is a string, it will be treated as a column name

Reply:
wait for a moment, let me check whether those expression usually accept literal strings

Reply:
I checked 

`to_char` : https://github.com/apache/spark/blob/3433f2a77d3dd665f42aa3d558152cf4c912c54c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/numberFormatExpressions.scala#L227-L234

`to_binary`:
https://github.com/apache/spark/blob/f718b025d87ae3726210c60ff71cb34917b32f51/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala#L2727-L2734

the format should be a literal string, so on the python side, the augment type should be `Union[Column, str]`, when the input is a `str`, convert it to `lit(...)`


also cc @HyukjinKwon

Reply:
I am fine with several exceptions here.

Reply:
The problem here is that it becomes a bit ambiguous if we support to take actual columns instead of just literals in `col2` and `col3`, which actually happens time to time. Or, e.g., if `col2` and `col3` support other types. Maybe we should just keep the signature as `ColumnOrName`.

`sentences` is one example.

Reply:
sounds good, so let us use `ColumnOrName`, which treats `str` arguments as column names. @panbingkun 

another advantage is that it can support non-foldable expressions in the future

Reply:
OK，Let me fix it.